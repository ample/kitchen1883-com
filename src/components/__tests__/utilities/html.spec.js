import React from "react"
import renderer from "react-test-renderer"

import HTML from "../../utilities/html"

const validNode = {
  childMarkdownRemark: {
    html: "<p>Hello World</p>",
  },
}

describe("HTML", () => {
  it("renders null for empty objects", () => {
    const tree = renderer.create(<HTML field={{}} />).toJSON()
    expect(tree).toBeNull()
    expect(tree).toMatchSnapshot()
  })
  it("renders null for incorrectly structured objects", () => {
    const tree = renderer.create(<HTML field={{ hello: "world" }} />).toJSON()
    expect(tree).toBeNull()
    expect(tree).toMatchSnapshot()
  })
  it("renders correctly", () => {
    const tree = renderer.create(<HTML field={validNode} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
