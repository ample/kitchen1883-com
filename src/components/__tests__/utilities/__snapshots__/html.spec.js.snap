// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTML renders correctly 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>Hello World</p>",
    }
  }
/>
`;

exports[`HTML renders null for empty objects 1`] = `null`;

exports[`HTML renders null for incorrectly structured objects 1`] = `null`;
